### PR TITLE
Documentation update for RDC

### PR DESCRIPTION
**Pull request by amullick007**

### CODE REVIEW

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:739]: All table numbers defined later in the spec should be incremented to account for new tables here.

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:799]: Nit: "therby" should be "thereby"

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:802]: "therby"

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:821]: I think we're generally trying to avoid 1st person pronouns in the spec - can this be reworded as generic implementation point of view instead of "we/our"?

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:804]: Could we add description of the methodology for forcing AHB bus to quiesce on occurrence of generated resets? I.e., use of reset windows as in https://github.com/chipsalliance/caliptra-rtl/blob/839800557abf621909d95e752553a52cd0957008/src/integration/rtl/caliptra_top.sv#L328

**Review by calebofearth — CHANGES_REQUESTED** [review]
Can you add tool versions for RDC analysis to the README?

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:739]: done, please check

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md]: corrected, thanks

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md]: corrected, thanks

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md]: reworded, please check

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md]: added details about the process of generating window, please check

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [reply docs/CaliptraIntegrationSpecification.md]: Looks great, thanks

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:859]: Should this also be a numbered table?
We've also been captioning and numbering Figures in the document  - could you make that update as well for new images?

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:804]: Typo: "firware"

File: docs/CaliptraIntegrationSpecification.md
Comment (calebofearth) [inline docs/CaliptraIntegrationSpecification.md:816]: Typo: "Constraint" should be "Constrain"

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:844]: Minor updates to improve accessibility, clarity, and update article: `In the preceding schematic, an RDC crossing violation is identified at the *data_q* flop, which is located on the right side of the figure.`

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:846]: remove `-` at end of intro sentence

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:857]: Update: The following scenarios can occur.

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:861]: Updates for present tense: `will prevent the flop to be loaded.`--`prevents the flop from loading.`

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:862]: `mux will switch to the leg which is on the same reset domain`-->`mux switches to the leg that is on the same reset domain`

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:863]: Updates for present tense and clarity: RDC violations are suppressed because fw_ov_fifo_wr_pulse = 0 (the red data lines on uc_rst are not able to reach data_q because load_data is 0)

File: docs/CaliptraIntegrationSpecification.md
Comment (steph-morton) [inline docs/CaliptraIntegrationSpecification.md:865]: Minor updates: Referring to the preceding table, if it can be ensured that *fw_ov_fifo_wr_pulse* is 0 when *cptra_uc_rst_b* is asserted, RDC crossings (Case #3) can be avoided. This condition is described in the preceding Constraints section.

**Review by steph-morton — COMMENTED** [review]
Edits for style, clarity, punctuation, and typos.

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:804]: thanks, updated. Quiesce is a more technical term which is used to indicate whenever an interface is idle'ed.

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:859]: thanks, done

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:804]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:816]: thanks, corrected

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:734]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:739]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:737]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:769]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:770]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:775]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:783]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:789]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:794]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:797]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:798]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:799]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:804]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:821]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:834]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:844]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:846]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:857]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:862]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:863]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:865]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:861]: thanks, updated

File: docs/CaliptraIntegrationSpecification.md
Comment (amullick007) [reply docs/CaliptraIntegrationSpecification.md:792]: thanks, updated

**Review by calebofearth — APPROVED** [review]

**Review by steph-morton — DISMISSED** [review]

**Comment by calebofearth** [comment]
@amullick007 looks like you'll need to merge from main to resolve the conflict on pr_hash/pr_timestamp.
I'll see if I can make this requirement a bit less onerous as we work towards 2.0.

**Comment by amullick007** [comment]
Hi Caleb,

Do I need write permissions to resolve merge conflicts?

"Only those with write access<https://docs.github.com/articles/what-are-the-different-access-permissions> to this repository can merge pull requests."

[cid:3a35b9c9-4448-41f0-858c-7bfb8e0b58ac]

Regards,
Avirup
________________________________

**Comment by amullick007** [comment]
> @amullick007 looks like you'll need to merge from main to resolve the conflict on pr_hash/pr_timestamp. I'll see if I can make this requirement a bit less onerous as we work towards 2.0.

I resolved the merge conflicts. Since I am not very familiar with Git, it took me few iterations to do this. Can you please check if the changes look good?

**Comment by calebofearth** [comment]
> > @amullick007 looks like you'll need to merge from main to resolve the conflict on pr_hash/pr_timestamp. I'll see if I can make this requirement a bit less onerous as we work towards 2.0.
> 
> I resolved the merge conflicts. Since I am not very familiar with Git, it took me few iterations to do this. Can you please check if the changes look good?

Merge looks a little funky, but seems like you caught all the file updates so I think it's good. 
Now that the fileset is updated you'll need to rerun stamp_repo.sh though, as the file hash has changed. Committing those updates shouldn't have any conflicts now.

**Review by calebofearth — DISMISSED** [review]

**Review by calebofearth — APPROVED** [review]

**Comment by amullick007** [comment]
All checks have passed, @steph-morton , @Nitsirks Can you please approve this PR.

**Review by Nitsirks — APPROVED** [review]

**Comment by amullick007** [comment]
> @amullick007 any idea what's going on with all these merge-base messages? It seems like approvals are being auto-dismissed for no reason.

no idea :( shall I create a new PR? Even though I am seeing that only 1 pending reviewer is left. Can @steph-morton approve the review and lets see it it allows to merge

**Review by steph-morton — DISMISSED** [review]

**Comment by calebofearth** [comment]
> > @amullick007 any idea what's going on with all these merge-base messages? It seems like approvals are being auto-dismissed for no reason.
> 
> no idea :( shall I create a new PR? Even though I am seeing that only 1 pending reviewer is left. Once @steph-morton approves the review, lets see it it allows to merge

We may not need to abandon this PR just yet.
Give this set of command line operations a try. Assuming you have your forked repo checked out locally somewhere under the remote name 'origin' (which is default):
```
git switch main
git pull --ff-only
git reset --hard 7761460547e3cc36e59a6ecad457c3007f49fac7
git merge --no-commit --squash 40bd9eb6367a8cb3129074d4b951c1930952f84a
git commit -m "Squash replay of RDC documentation updates"
bash .github/scripts/stamp_repo.sh
git commit -m "Update stamps" -- .github/workflow_metadata/pr_hash .github/workflow_metadata/pr_timestamp
git push -f origin main:main
```
This should essentially start over from scratch and replay all your edits onto this same branch, then re-stamp it.
This will revert the merge from the chipsalliance main branch, which presumably is causing these merge-base issues.

**Review by calebofearth — APPROVED** [review]

**Review by Nitsirks — APPROVED** [review]

**Comment by calebofearth** [comment]
Thanks for the fixups Avirup. Looks like it's good now. I'll merge once the checks finish.

**Pull request merged**
